File: sdk/python/x402_client.py
import requests

class X402Client:
    def __init__(self, base_url: str):
        self.base_url = base_url.rstrip("/")

    def supported(self):
        r = requests.get(f"{self.base_url}/supported")
        r.raise_for_status()
        return r.json()

    def price(self, quote: dict):
        r = requests.post(f"{self.base_url}/price", json=quote)
        if r.status_code != 402:
            raise RuntimeError(f"expected 402, got {r.status_code}")
        return r.json()

    def verify(self, payload: dict):
        r = requests.post(f"{self.base_url}/verify", json=payload)
        r.raise_for_status()
        return r.json()

    def verify_settlement(self, payload: dict):
        r = requests.post(f"{self.base_url}/verify-settlement", json=payload)
        r.raise_for_status()
        return r.json()

    def attest(self, att: dict):
        r = requests.post(f"{self.base_url}/attest", json=att)
        r.raise_for_status()
        return r.json()

File: sdk/node/x402Client.ts
import { ethers } from "ethers";

export type FacilitatorConfig = {
  baseUrl: string;
  gateway: string;
  provider: ethers.Provider;
  signer: ethers.Signer;
};

export class X402Client {
  constructor(private readonly cfg: FacilitatorConfig) {}

  async supported() {
    const r = await fetch(`${this.cfg.baseUrl}/supported`);
    if (!r.ok) throw new Error(`supported failed: ${r.status}`);
    return r.json();
  }

  async price(quote: any) {
    const r = await fetch(`${this.cfg.baseUrl}/price`, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(quote)
    });
    if (r.status !== 402) throw new Error(`expected 402, got ${r.status}`);
    const body = await r.json();
    return body;
  }

  async verify(payload: any) {
    const r = await fetch(`${this.cfg.baseUrl}/verify`, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(payload)
    });
    return r.json();
  }

  async verifySettlement(payload: any) {
    const r = await fetch(`${this.cfg.baseUrl}/verify-settlement`, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(payload)
    });
    return r.json();
  }

  async attest(att: any) {
    const r = await fetch(`${this.cfg.baseUrl}/attest`, {
      method: "POST",
      headers: { "content-type": "application/json" },
      body: JSON.stringify(att)
    });
    if (!r.ok) throw new Error(`attest failed: ${r.status}`);
    return r.json();
  }
}
